src/libs/antares/study/area/links.cpp

```diff
@@ -337,38 +346,58 @@ bool AreaLinksLoadFromFolder(Study& study, AreaList* l, Area* area, const AnyStr
 
         if (study.header.version < 320)
         {
-            link.data.resize(fhlMax, HOURS_PER_YEAR);
-            link.data.zero();
-            if (link.data.jit)
+            // gp : refactor this part : avoid code duplication
+            link.parameters.resize(fhlMax, HOURS_PER_YEAR);
+            link.parameters.zero();
+            if (link.parameters.jit)
+            {
+                link.parameters.jit->options |= Matrix<>::optFixedSize;
+                link.parameters.markAsModified();
+            }
+
+            link.directCapacities.resize(1, HOURS_PER_YEAR);
+            link.directCapacities.zero();
+            if (link.directCapacities.jit)
+            {
+                link.directCapacities.jit->options |= Matrix<>::optFixedSize;
+                link.directCapacities.markAsModified();
+            }
+
+            link.indirectCapacities.resize(1, HOURS_PER_YEAR);
+            link.indirectCapacities.zero();
+            if (link.indirectCapacities.jit)
             {
-                link.data.jit->options |= Matrix<>::optFixedSize;
-                link.data.markAsModified();
+                link.indirectCapacities.jit->options |= Matrix<>::optFixedSize;
+                link.indirectCapacities.markAsModified();
             }
 
             Matrix<double> tmp;
-            // NTC
+            // NTC direct
             buffer.clear() << folder << SEP << link.with->id << SEP << "direct."
                            << study.inputExtension;
             ret = tmp.loadFromCSVFile(
-                    buffer, 1, 8760, Matrix<>::optFixedSize | Matrix<>::optImmediate)
+                    buffer, 1, HOURS_PER_YEAR, Matrix<>::optFixedSize | Matrix<>::optImmediate)
                   && ret;
-            link.data.pasteToColumn(fhlNTCDirect, tmp[0]);
+            link.directCapacities.pasteToColumn(0, tmp[0]);
```

Comment:
There is no need to use a buffer. Load the file contents into `directMatrix` directly.

It wasn't possible previously, but now it is.